bagger/src/bag_assembly.py

```diff
@@ -67,4 +62,5 @@ def get_separator():
 def ensure_directory(destination):
     destination_dir = os.path.dirname(destination)
     if not os.path.exists(destination_dir):
-        os.makedirs(destination_dir)  # TODO: beware race condition if threaded
+        os.makedirs(destination_dir)
+        # TODO: beware race condition if threaded
```

Comment:
minor: not sure why we're fiddling with this code, but if you are, you might change this to

```python
os.makedirs(destination_dir, exist_ok=True)
```

which is race-free and simpler!